### DISCUSSION TITLE
Offline / Online sync strategy

### DISCUSSION THREAD

**Discussion by moshiki** [flutterdata/flutter_data#203] [Q&A]
Hi, I am wondering about the strategy used for data sync with the REST API. 
In our scenario the data on the server side can be altered through multiple endpoints. Will the data just be overridden on the REST API as soon as the device has online access again, or is there a way to determine the newest piece of data and combine the datasets (server side has altered in meantime and is newer now)?

**Comment by frank06** [comment]
Yes, you should perform the reconciliation manually. Another hook would be needed on FD, something like `onRetry` before the operation is reattempted (which gives you a chance to pull data and sync)

**Comment by peppius** [comment]
Hi, instead I ask if the device is offline when save data, data sync automatically with REST API when the device comes back online? Or I will sync data manually?

**Reply by frank06** [reply]
Yes, for now it's done automatically.

FD does not distinguish between online and offline – the only thing that matters is whether a request was successful or not. If the failure was due to connectivity reasons, it will be retried.
